internal/pkg/manifest/pipeline_test.go

```diff
@@ -261,6 +261,48 @@ stages:
 				},
 			},
 		},
+		"valid pipeline.yml with build": {
```

Comment:
Thank you for writing all these tests! They are really helpful and comprehensive.